test/join-wrapper-mocha.js

```diff
@@ -0,0 +1,349 @@
+                    expect(done.groupLm).to.be.undefined;
+                    done.lm.match(/^LM(\d+)\.(\d+)$/).should.have.length(3);
+                });
+            });
+
+            it("should not trigger updater function with no input data", function() {
+                var node =
+                    test.createComponent(
+                        joinWrapper({
+                                     inPorts:{
+                                         hash: {required: true},
+                                     },
+                                     updater: function() {
+                                         throw Error('updater ran');
+                                     }
+                                  })
+                   );
+
+                return new Promise(function(done, fail){
+
+                    // Fail if we get output or error:
+                    test.onOutPortData(node, 'output', fail);
+                    test.onOutPortData(node, 'error', fail);
+
+                    // Send hash, but no input data to join
+                    test.sendData(node, 'hash', {"1":"one"});
+
+                    setTimeout(function() { 
+                        done();
+                    },200);
+                });
+            });
+
+            it("should call updater with VNI context", function() {
+                var handler;
+                var updater = function(hash, input) {
+                   this.should.have.all.keys('vnid', 'delete', 'inputStates',
+                                              'errorState', 'outputState', 'nodeInstance');
+                   handler('success');
+                };
+                var node =
+                    test.createComponent(joinWrapper(updater));
+   
+                return new Promise(function(callback){
+                    handler = callback;
+                    var lm = createLm();
+                    test.sendData(node, 
+                                  'hash', 
+                                  {vnid: '', data: {"1": "one"}, 
+                                   error:undefined, stale: undefined, 
+                                   groupLm: undefined, lm: lm});
+                    test.sendData(node, 
+                                  'input', 
+                                  {vnid: '1', data: "ONE", 
+                                   error:undefined, stale: undefined, 
+                                   groupLm: lm, lm: createLm()});
+                }).should.become('success');
+            });
+        });
+
+        describe('functional behavior', function() {
+
+            it("should execute in a noflo network", function() {
+                this.timeout(4000);
+                return test.createNetwork(
+                    { nodeA: 'rdf-components/parse-json',
+                      nodeS: 'rdf-components/simple-splitter',
+                      nodeB: 'rdf-components/to-upper-case',
+                      nodeJ: { getComponent:
```

Comment:
You can pass the function directly, no need to wrap it in a getComponent property.